pkg/controller/handler.go

```diff
@@ -137,6 +138,50 @@ func (h *handler) CreateServiceInstance(in *servicecatalog.Instance) (*serviceca
 	return h.apiClient.Instances(in.ObjectMeta.Namespace).Update(in)
 }
 
+func (h *handler) DeleteServiceBinding(sb *servicecatalog.Binding) error {
+	// this logic to set and update the timestamp is TPR specific. to be moved to the API server
+	dts := metav1.Now()
+	sb.DeletionTimestamp = &dts
+	if _, err := h.apiClient.Bindings(sb.Namespace).Update(sb); err != nil {
+		return err
+	}
+
+	// uninject
+	if err := h.injector.Uninject(sb); err != nil {
+		// if 0 conditions, uninject and drop condition for uninject
+		// TODO: add failure condition (https://github.com/kubernetes-incubator/service-catalog/issues/305)
+		return err
+	}
+	// TODO: add success condition (https://github.com/kubernetes-incubator/service-catalog/issues/305)
+	if _, err := h.apiClient.Bindings(sb.Namespace).Update(sb); err != nil {
+		return err
+	}
+
+	// TODO: unbind && add conditions (https://github.com/kubernetes-incubator/service-catalog/issues/305)
+	if err := h.unbind(sb); err != nil {
+		// TODO: add failure condition (https://github.com/kubernetes-incubator/service-catalog/issues/305)
+		return err
+	}
+	// TODO: add success condition (https://github.com/kubernetes-incubator/service-catalog/issues/305)
+
+	if _, err := h.apiClient.Bindings(sb.Namespace).Update(sb); err != nil {
```

Comment:
@rifung you're right that `sb` isn't modified after the last `Update` and this, but after the above `TODO` is completed, it will be. @bmelville were writing this quickly at the boulder face-to-face, so we intended for that to be in a follow-up PR